packages/preview-middleware-client/src/cpe/connector-service.ts

```diff
@@ -12,7 +13,7 @@ export class WorkspaceConnectorService {
      * @param sendAction action sender function
      */
     public async init(sendAction: ActionSenderFunction): Promise<void> {
-        const { version } = (await VersionInfo.load()) as { version: string };
+        const version = (await VersionInfo.load() as VersionInformation)?.libraries?.find((lib) => lib.name === 'sap.ui.core')?.version ?? '';
```

Comment:
`as string` or  `?? ''` which one is to be preferred? 🤔

Reply:
as string is wrong because it could be undefined, so ?? '' is the better option

Reply:
If `undefined` is a real option there's quite some more to be adjusted. See [here](https://github.com/SAP/open-ux-tools/blob/feat/1679/preview20/packages/preview-middleware-client/src/adp/init.ts#L21) or [here](https://github.com/SAP/open-ux-tools/blob/feat/1679/preview20/packages/preview-middleware-client/src/cpe/connector-service.ts#L17) or [here](https://github.com/SAP/open-ux-tools/blob/feat/1679/preview20/packages/preview-middleware-client/src/cpe/outline/nodes.ts#L92) or ... all of these ignore a potentially `undefined` `version` property.

Good thing is the type checker finds all of them now by adjusting [SingleVersionInfo](https://github.com/SAP/open-ux-tools/blob/feat/1679/preview20/packages/preview-middleware-client/types/global.d.ts#L7) to be `| undefined` which is anyway a better type according to [sap/ui/VersionInfo.load](https://github.com/SAP/openui5/blob/master/src/sap.ui.core/src/sap/ui/VersionInfo.js#L40).

Errors  Files
     2  src/adp/init.ts:21
     1  src/cpe/connector-service.ts:17
     1  src/cpe/outline/nodes.ts:92
     1  src/flp/init.ts:281
     1  src/flp/WorkspaceConnector.ts:74

Reply:
I created a new issue for that one to be done in an other PR at a later point in time: https://github.com/SAP/open-ux-tools/issues/2207